### PR TITLE
fix(footer): set max-toolbar-items when mini table is used

### PR DESCRIPTION
**Pull request by micaelagoffe**
Set max-toolbar-items on cosmoz-bottom-bar when mini table is used

[AB#17737](https://dev.azure.com/neovici/Cosmoz3/_backlogs/backlog/UX/Stories/?workitem=17737)

### CODE REVIEW

File: lib/use-footer.js [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
-		} = host;
+		} = host,
+		isMiniTable = host.hasAttribute('mini');
 
 	return {
 		csvFilename,
 		xlsxFilename,
 		xlsxSheetname,
 		topPlacement,
+		isMiniTable,
```
Comment (cristinecula) [inline lib/use-footer.js:18]:
```suggestion
		maxToolbarItems,
```

File: lib/use-footer.js [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
-}) => {
+export const useFooter = ({ host, ...rest }) => {
 	const {
 			csvFilename = 'omnitable.csv',
 			xlsxFilename = 'omnitable.xlsx',
 			xlsxSheetname = 'Omnitable',
 			topPlacement = _defaultPlacement,
-		} = host;
+		} = host,
+		isMiniTable = host.hasAttribute('mini');
```
Comment (cristinecula) [inline lib/use-footer.js:11]: This is not a great way to determine if the omnitable is in `mini` state. You should make use of [isMiniSize](https://github.com/Neovici/cosmoz-omnitable/blob/18c0da7d4dc8ffa8e11708f0fd529ab0271715b5/lib/use-mini.js#L5) and determine the number of toolbar items in the main hook.

```suggestion
      maxToolbarItems,
		} = host;
```

**Review by cristinecula — CHANGES_REQUESTED** [review]

**Comment by cristinecula** [comment]
Ideally the cosmoz-bottom-bar element would use intersection observers or some other mechanism to determine if it should show/hide the button.

File: lib/use-omnitable.js [hunk 3]
```diff
@@ -40,14 +40,15 @@ export const useOmnitable = (host) => {
...
-		{ collapsedColumns, miniColumns } = useFastLayout({
+		{ isMiniSize, collapsedColumns, miniColumns } = useFastLayout({
 			host,
 			columns,
 			settings,
 			setSettings,
 			resizeSpeedFactor,
 			sortAndGroupOptions,
 		}),
+		maxToolbarItems = host.hasAttribute('mini') && isMiniSize ? '0' : '',
```
Comment (cristinecula) [inline lib/use-omnitable.js:51]: Reading the `isMini` state from the `host` element is too indirect and will lead to confusion. Instead, you should alter the code in  `useMini` to provide this information as a flag.

https://github.com/Neovici/cosmoz-omnitable/blob/18c0da7d4dc8ffa8e11708f0fd529ab0271715b5/lib/use-mini.js#L3-L29

```js
  const hasMiniColumn =  !!miniColumn

	useEffect(() => {
		host.toggleAttribute('mini', hasMiniColumn);
	}, [hasMiniColumn]);

  return {
    isMini: hasMiniColumn && isMiniSize
  }
```

Comment (cristinecula) [reply lib/use-omnitable.js:51]:
```suggestion
		maxToolbarItems = isMini ? 0 : undefined, // should be a number or undefined
```

File: lib/render-footer.js [hunk 4]
```diff
@@ -11,61 +11,56 @@ export const renderFooter = ({
...
-			</slot>
-			<slot name="actions" id="actions"></slot>
-			<!-- These slots are needed by cosmoz-bottom-bar
+	topPlacement,
+	maxToolbarItems,
+}) =>
+	html`<cosmoz-bottom-bar
+		id="bottomBar"
+		?active=${!isEmpty(selectedItems.length)}
+		?max-toolbar-items=${!isEmpty(maxToolbarItems)}
```
Comment (cristinecula) [inline lib/render-footer.js:20]: `?prop` should only be used for boolean attributes, for number props you should use `.prop`. 

```suggestion
		.maxToolbarItems=${maxToolbarItems}
```

**Review by cristinecula — CHANGES_REQUESTED** [review]

File: lib/use-omnitable.js (hunk 3 above)
Comment (micaelagoffe) [reply lib/use-omnitable.js]: After our discussion in slack I'm using ´'0' : undefined´, so I can keep the isEmpty check in render-footer.

File: lib/render-footer.js (hunk 4 above)
Comment (micaelagoffe) [reply lib/render-footer.js:20]: Keeping this change after our discussion in slack 🙂

**Review by cristinecula — APPROVED** [review]

**Pull request merged**